fiftyone/factory/repos/delegated_operation.py

```diff
@@ -210,6 +224,45 @@ def update_run_state(
         if update is None:
             raise ValueError("Invalid run_state: {}".format(run_state))
 
+        if progress is not None:
+            update["$set"]["status"] = progress
+            update["$set"]["status"]["updated_at"] = datetime.utcnow()
+
+        doc = self._collection.find_one_and_update(
+            filter={"_id": _id},
+            update=update,
+            return_document=pymongo.ReturnDocument.AFTER,
+        )
+
+        return DelegatedOperationDocument().from_pymongo(doc)
+
+    def update_progress(
+        self,
+        _id: ObjectId,
+        progress: ExecutionProgress,
+    ) -> DelegatedOperationDocument:
+        execution_progress = progress
+        if not isinstance(progress, ExecutionProgress):
+            if isinstance(progress, dict):
+                execution_progress = ExecutionProgress(**progress)
+            else:
+                raise ValueError("Invalid progress: {}".format(progress))
+
+        if not execution_progress or (
+            not execution_progress.progress and not execution_progress.label
```

Comment:
Based on an offline discussion
```suggestion
            execution_progress.progress is None and not execution_progress.label
```